### DISCUSSION TITLE
Hide device groups & machines from users that do not have access.

### DISCUSSION THREAD

**Discussion by ok1ok1** [immense/Remotely#528] [Q&A]
I'm really liking Remotely but I hate the fact that all users regardless of device group permissions are able to see the the device groups and specs/ID/IP of all client devices. Connections to these devices are not permitted, why are they shown at all? This seems very odd. Maybe I'm missing something. Is there a setting to change this behavior?

**Comment by AMotaPT** [comment]
I create a Issue Post also about that, the user can't control devices but can change to his group, save, and now can control it.
Something is missing for sure.

**Comment by AMotaPT** [comment]
@ok1ok1 
New update available:
https://github.com/immense/Remotely/releases/tag/v2023.02.01.2115

**Comment by PCProsCA** [comment]
This is a weird design decision, however it can be resolved in the following way.
_TL/DR: make sure all devices are in a Device Group._

Log in as admin, and for every device, follow these steps:
Assuming you have already created user(s)...
Go to Organization
Add new **Device Group**: use [COMPUTERNAME] (thinking of Device Groups like individual computers seems to work best for me)
Next, Edit the user that would have access to this machine and check the box for the Device Group(s) they have access to.
Doing it this way, any user can have access to any subset of computers.
Finally, I log in to the target PC and download the Resident Agent from your remotely site.  Leave it in the downloads folder, run CMD as admin, and use the following script to deploy:

%HOMEPATH%\Downloads\Remotely_Install-[227b].exe -install -quiet -organizationid "YOUR-ORG-ID" -serverurl "https://YOUR-SERVER-URL/" -devicegroup "YOUR-MACHINE-GROUP"

for example:
%HOMEPATH%\Downloads\Remotely_Install-[227b].exe -install -quiet -organizationid "8db55a17-d2d3-4d7f-9c73-13098123098" -serverurl "https://remotely.blah.com/" -devicegroup "DESKTOP1"

In a few moments, you should see the device appear in your Remotely site and only folks who have access to that device group will be able to see the device.

**If you create a device not in a device group, ALL REMOTELY USERS will be able to see it!**

Hope that helps!

Ryan

**Comment by bitbound** [comment]
The groups/permissions system needs to be completely rewriten.

When I first started the project (back when it was DoXM), groups and permissions didn't exist. I built everything from the perspective of a single tech-savvy person using it to manage family and friends' computers.  I wasn't originally anticipating that it would be used in a shared professional environment.

At some point, I tacked on groups in a hasty fashion, and the implementation was kinda garbage.  So I apologize for that.  😊

**Comment by PCProsCA** [comment]
No problem, Jared.
We’re making it work!
Thanks for all your hard work over the years.

**Reply by bitbound** [reply]
> No problem, Jared. We’re making it work! Thanks for all your hard work over the years.

This comment made my day!  Thank you.  :)
